meta-ostro/classes/ostro-image.bbclass

```diff
@@ -10,121 +10,245 @@ IMAGE_INSTALL = " \
+#   BB_ENV_EXTRAWHITE="$BB_ENV_EXTRAWHITE OS_VERSION" OS_VERSION=110 bitbake ostro-image-swupd
+#   ...
+
+# Define additional bundles. This matches 1:1 to image features
+# which add packages (i.e. OSTRO_IMAGE_PKG_FEATURES).
+# In addition, for each of these we also create a development bundle
+# that also contains the development files.
+#SWUPD_BUNDLES ??= " \
+#    ${OSTRO_IMAGE_PKG_FEATURES} \
+#    ${@ ' '.join([x + '-dev' for x in '${OSTRO_IMAGE_PKG_FEATURES}'.split()])} \
+#"
+BUNDLE_CONTENTS[can] = "${FEATURE_PACKAGES_can}"
+BUNDLE_CONTENTS[connectivity] = "${FEATURE_PACKAGES_connectivity}"
+BUNDLE_CONTENTS[devkit] = "${FEATURE_PACKAGES_devkit}"
+BUNDLE_CONTENTS[iotivity] = "${FEATURE_PACKAGES_iotivity}"
+BUNDLE_CONTENTS[java-jdk] = "${FEATURE_PACKAGES_java-jdk}"
+BUNDLE_CONTENTS[node-runtime] = "${FEATURE_PACKAGES_node-runtime}"
+BUNDLE_CONTENTS[nodejs-runtime-tools] = "${FEATURE_PACKAGES_nodejs-runtime-tools}"
+BUNDLE_CONTENTS[python-runtime] = "${FEATURE_PACKAGES_python-runtime}"
+BUNDLE_CONTENTS[qatests] = "${FEATURE_PACKAGES_qatests}"
+BUNDLE_CONTENTS[ssh-server] = "${FEATURE_PACKAGES_ssh-server-openssh}"
+BUNDLE_CONTENTS[soletta] = "${FEATURE_PACKAGES_soletta}"
+BUNDLE_CONTENTS[soletta-tools] = "${FEATURE_PACKAGES_soletta-tools}"
+BUNDLE_CONTENTS[tools-debug] = "${FEATURE_PACKAGES_tools-debug}"
+BUNDLE_CONTENTS[tools-develop] = "${FEATURE_PACKAGES_tools-develop}"
+BUNDLE_CONTENTS[tools-interactive] = "${FEATURE_PACKAGES_tools-interactive}"
+
+# Defining bundles as above is currently too slow (build times in the CI
+# of more than three hours despite reused sstate cache). Let's cut down
+# the number of bundles to something more manageable and increase it again
+# after improving bundle creation performance.
+SWUPD_BUNDLES ??= " \
+    reference \
+    full \
+    full-dev \
+"
+BUNDLE_CONTENTS[reference] = " \
+    ${FEATURE_PACKAGES_connectivity} \
+    ${FEATURE_PACKAGES_tools-interactive} \
+    ${FEATURE_PACKAGES_ssh-server-openssh} \
+"
+BUNDLE_CONTENTS[full] = " \
+    ${FEATURE_PACKAGES_can} \
+    ${FEATURE_PACKAGES_devkit} \
+    ${FEATURE_PACKAGES_iotivity} \
+    ${FEATURE_PACKAGES_java-jdk} \
+    ${FEATURE_PACKAGES_node-runtime} \
+    ${FEATURE_PACKAGES_python-runtime} \
+    ${FEATURE_PACKAGES_qatests} \
+    ${FEATURE_PACKAGES_soletta} \
+    ${FEATURE_PACKAGES_soletta-tools} \
+    ${FEATURE_PACKAGES_tools-debug} \
+    ${FEATURE_PACKAGES_tools-develop} \
+    ${FEATURE_PACKAGES_tools-interactive} \
+"
+
+# When swupd bundles are enabled, choose explicitly which images
+# are created. The base image will only have the core-os bundle and
+# thus might not be very useful. For use in ostro-image-swupd.bb we
+# pre-define additional images:
+# ostro-image-swupd-reference -
+#    Base image plus login via getty and ssh, plus connectivity.
+#    This is what developers  are expected to start with when
+#    building their first image.
```

Comment:
So following that logic, ostro-image.bbclass as the base class for custom images shouldn't contain any bundle definitions, right? That would be in the OEM's swupd image, which would be ostro-image-swupd-reference.bb in our case.

If we change the base name like this, the main image will still be called "ostro-image-swupd-reference". But bundle images with additional bundles become ostro-image-swupd-reference-<foo>, for example ostro-image-swupd-reference-dev. Is that a good or bad thing?

I'm undecided myself. It's consistent and keeps the image name space clean, but it's also a bit long.